pkg/bosh/manifest/dns.go

```diff
@@ -0,0 +1,323 @@
+			var a = make([]Alias, 0)
+			err = json.Unmarshal(aliasesBytes, &a)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Loading aliases from manifest")
+			}
+			dns.aliases = append(dns.aliases, a...)
+		}
+	}
+	return &dns, nil
+}
+
+// FindServiceNames see interface
+func (dns *boshDomainNameService) FindServiceNames(instanceGroupName string, deploymentName string) []string {
+	result := make([]string, 0)
+	for _, alias := range dns.aliases {
+		for _, target := range alias.Targets {
+			if target.InstanceGroup == instanceGroupName {
+				result = append(result, strings.Split(alias.Domain, ".")[0])
+			}
+		}
+	}
+	if len(result) == 0 {
+		result = append(result, serviceName(instanceGroupName, deploymentName, 63))
+	}
+	return result
+}
+
+// HeadlessServiceName see interface
+func (dns *boshDomainNameService) HeadlessServiceName(instanceGroupName string, deploymentName string) string {
+	serviceNames := dns.FindServiceNames(instanceGroupName, deploymentName)
+	if len(serviceNames) == 0 {
+		return serviceName(instanceGroupName, deploymentName, 63)
+	}
+	return serviceNames[0]
+}
+
+// ConfigurePod see interface
+func (dns *boshDomainNameService) ConfigurePod(podSpec *v1.PodSpec) {
+	if dns.localDNSIP == "" {
+		panic("BoshDomainNameService: ConfigurePod called before Reconcile")
+	}
+	podSpec.DNSPolicy = v1.DNSNone
+	podSpec.DNSConfig = &v1.PodDNSConfig{
+		Nameservers: []string{dns.localDNSIP},
+		Searches:    []string{fmt.Sprintf("%s.svc.cluster.local", dns.namespace), "service.cf.internal"},
+	}
+}
+
+// Reconcile see interface
+func (dns *boshDomainNameService) Reconcile(ctx context.Context, c client.Client, setOwner func(object v12.Object) error) error {
+
+	rewrites := ""
+	for _, alias := range dns.aliases {
+		for _, target := range alias.Targets {
+			serviceName := serviceName(target.InstanceGroup, dns.namespace, 63)
+			rewrites = rewrites + fmt.Sprintf("  rewrite name exact %s %s.%s.svc.cluster.local\n", serviceName, strings.Split(alias.Domain, ".")[0], dns.namespace)
+			rewrites = rewrites + fmt.Sprintf("  rewrite name exact %s.%s %s.%s.svc.cluster.local\n", serviceName, dns.namespace, strings.Split(alias.Domain, ".")[0], dns.namespace)
+		}
+	}
+	metadata := v12.ObjectMeta{
+		Name:      "bosh-dns",
+		Namespace: dns.namespace,
+		Labels:    map[string]string{"app": "bosh-dns"},
+	}
+
+	configMap := v1.ConfigMap{
+		ObjectMeta: metadata,
+		Data:       map[string]string{"Corefile": fmt.Sprintf(corefile, rewrites, dns.namespace, dns.rootDNSIP())},
+	}
+	service := v1.Service{
+		ObjectMeta: metadata,
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{Name: "dns", Port: 53, Protocol: "UDP", TargetPort: intstr.FromInt(8053)},
+				{Name: "dns-tcp", Port: 53, Protocol: "TCP", TargetPort: intstr.FromInt(8053)},
+				{Name: "metrics", Port: 9153, Protocol: "TCP", TargetPort: intstr.FromInt(9153)},
+			},
+			Selector: map[string]string{"app": "bosh-dns"},
+			Type:     "ClusterIP",
+		},
+	}
+
+	var mode int32 = 420
```

Comment:
This can be global.

Reply:
Why should it be global?
I'd even write it inline, but it's not possible as `&420` is not valid int literal and we need a pointer type.

Reply:
It _can_ be global, but not a must. This comment came out only because, in general terms, it just needs a single initialization, and it will never change. In C++, it would make sense to have it declared as `static const` right before its usage. Now that I think again, I shouldn't have pointed it out in the first place.